katz/src/main/kotlin/katz/effects/internal/AndThen.kt

```diff
@@ -0,0 +1,135 @@
+package katz.effects.internal
+
+/**
+ * Abandon all hope
+ * Ye who enter here
+ */
+internal sealed class AndThen<in A, out B> : (A) -> B {
```

Comment:
what we should do with this [andThen()](https://github.com/FineCinnamon/Katz/blob/master/katz/src/main/kotlin/katz/data/Reader.kt#L6)?